src/merge_maps_kinematic.cpp

```diff
@@ -44,61 +44,76 @@ void MergeMapTool::SetConfigs()
   {
     resolution_ = 0.05;
   }
-
   sstS_.push_back(nh_.advertise<nav_msgs::OccupancyGrid>("/map", 1, true));
   sstmS_.push_back(nh_.advertise<nav_msgs::MapMetaData>( \
                                                     "/map_metadata", 1, true));
-  ssMap_ = nh_.advertiseService("merge_maps", 
-                                        &MergeMapTool::MergeMapCallback, this);
-  ssSubmap_ = nh_.advertiseService("add_submap", 
-                                       &MergeMapTool::AddSubmapCallback, this);
+  ros::NodeHandle nh;
+  ssMap_ = nh.advertiseService("merge_submaps",
+                                        &MergeMapsKinematic::MergeMapCallback, this);
+  ssSubmap_ = nh.advertiseService("add_submap",
+                                       &MergeMapsKinematic::AddSubmapCallback, this);
   num_submaps_ = 0;
   tfB_ = new tf::TransformBroadcaster();
 }
 
 /*****************************************************************************/
-MergeMapTool::~MergeMapTool()
+MergeMapsKinematic::~MergeMapsKinematic()
 /*****************************************************************************/
 {
   if (interactive_server_)
   {
     delete interactive_server_;
   }
+  for(std::vector<karto::Dataset*>::iterator dataset_it = dataset_vec_.begin(); dataset_it != dataset_vec_.end(); ++dataset_it )
+  {
+    delete *dataset_it;
+  }
 }
 
 /*****************************************************************************/
-bool MergeMapTool::AddSubmapCallback(slam_toolbox::AddSubmap::Request &req,
+bool MergeMapsKinematic::AddSubmapCallback(slam_toolbox::AddSubmap::Request &req,
                                      slam_toolbox::AddSubmap::Response &resp)
 /*****************************************************************************/
 {
   // find if file exists
-  const std::string filename = req.filename + std::string(".st");
-  if (!FileExists(filename))
+  const std::string filename = req.filename;
```

Comment:
if you're not going to add an extension , delete the line and just pass into the function `req.filename`